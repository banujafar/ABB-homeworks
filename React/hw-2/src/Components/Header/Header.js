import { Component } from "react";
import { AiFillStar } from "react-icons/ai";
import {BsBasket2Fill} from 'react-icons/bs';
import styles from './Header.module.scss'
export class Header extends Component {
  constructor(props) {
    super(props);
  }
  render(){
    return(
        <ul className={styles["list-container"]}>
            <li className={styles["list-item"]}>
                <AiFillStar/>
                <span>Favorites: {this.props.favoritesCount}</span>
            </li>
            <li className={styles["list-item"]}>
                <BsBasket2Fill/>
                <span>Cart: {this.props.cartCount}</span>
            </li>
        </ul>
    )
  }
}
