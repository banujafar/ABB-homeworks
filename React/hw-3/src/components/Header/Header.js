import { AiFillStar } from "react-icons/ai";
import { BsBasket2Fill } from "react-icons/bs";
import styles from "./Header.module.scss";
import { Link } from "react-router-dom";
const Header = (props) => {
  return (
    <ul className={styles["list-container"]}>
      <Link to="/">
        <li className={styles["list-item"]}>
          <span>Home</span>
        </li>
      </Link>
      <Link to="/favorites">
        <li className={styles["list-item"]}>
          <AiFillStar />
          <span>Favorites: {props.favoritesCount}</span>
        </li>
      </Link>
      <Link to="/cart">
        <li className={styles["list-item"]}>
          <BsBasket2Fill />
          <span>Cart: {props.cartCount}</span>
        </li>
      </Link>
    </ul>
  );
};
export default Header;
