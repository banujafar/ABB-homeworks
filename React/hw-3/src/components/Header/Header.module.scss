.list-container {
  display: flex;
  justify-content: flex-end;
  align-items: center;

  .list-item {
    display: flex;
    justify-content: center;
    align-items: center;
    list-style: none;
    margin: 10px;
  }
}
