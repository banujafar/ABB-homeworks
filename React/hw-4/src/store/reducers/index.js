import { combineReducers } from "redux";
import { modalReducer } from "./modalReducer";
import { productReducer } from "./productReducer";
import { cartReducer } from "./cartReducer";
import { favoritesReducer } from "./favoritesReducer";

export const rootReducer=combineReducers({
    modal:modalReducer,
    products:productReducer,
    cart:cartReducer,
    favorites:favoritesReducer
})