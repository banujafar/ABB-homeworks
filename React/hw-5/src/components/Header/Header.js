import { AiFillStar } from "react-icons/ai";
import { BsBasket2Fill } from "react-icons/bs";
import styles from "./Header.module.scss";
import { Link } from "react-router-dom";
import { useSelector } from "react-redux";
const Header = () => {
  const { count } = useSelector((state) => state.cart);
  const favorites = useSelector((state) => state.favorites.favorites);

  return (
    <ul className={styles["list-container"]}>
      <Link to="/">
        <li className={styles["list-item"]}>
          <span>Home</span>
        </li>
      </Link>
      <Link to="/favorites">
        <li className={styles["list-item"]}>
          <AiFillStar />
          <span>Favorites: {favorites.length}</span>
        </li>
      </Link>
      <Link to="/cart">
        <li className={styles["list-item"]}>
          <BsBasket2Fill />
          <span>Cart: {count}</span>
        </li>
      </Link>
    </ul>
  );
};
export default Header;
