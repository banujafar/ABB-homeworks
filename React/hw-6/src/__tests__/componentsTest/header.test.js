import Header from "../../components/Header/Header";
import { Provider } from "react-redux";
import { store } from "../../store";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

test("should  matches DOM snapshot correctly", () => {
  const { container } = render(
    <BrowserRouter>
      <Provider store={store}>
        <Header />
      </Provider>
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});
