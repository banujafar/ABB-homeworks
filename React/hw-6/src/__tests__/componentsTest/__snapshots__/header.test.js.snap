// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should  matches DOM snapshot correctly 1`] = `
<div>
  <ul
    class="list-container"
  >
    <a
      href="/"
    >
      <li
        class="list-item"
      >
        <span>
          Home
        </span>
      </li>
    </a>
    <a
      href="/favorites"
    >
      <li
        class="list-item"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
          />
        </svg>
        <span>
          Favorites: 
          0
        </span>
      </li>
    </a>
    <a
      href="/cart"
    >
      <li
        class="list-item"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.929 1.757a.5.5 0 1 0-.858-.514L2.217 6H.5a.5.5 0 0 0-.5.5v1a.5.5 0 0 0 .5.5h.623l1.844 6.456A.75.75 0 0 0 3.69 15h8.622a.75.75 0 0 0 .722-.544L14.877 8h.623a.5.5 0 0 0 .5-.5v-1a.5.5 0 0 0-.5-.5h-1.717L10.93 1.243a.5.5 0 1 0-.858.514L12.617 6H3.383L5.93 1.757zM4 10a1 1 0 0 1 2 0v2a1 1 0 1 1-2 0v-2zm3 0a1 1 0 0 1 2 0v2a1 1 0 1 1-2 0v-2zm4-1a1 1 0 0 1 1 1v2a1 1 0 1 1-2 0v-2a1 1 0 0 1 1-1z"
          />
        </svg>
        <span>
          Cart: 
          0
        </span>
      </li>
    </a>
  </ul>
</div>
`;
